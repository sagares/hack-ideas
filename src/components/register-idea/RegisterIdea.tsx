import React, { useState } from "react";
import RegisterDialog from "./RegisterDialog";

import "./register-idea.less";

const RegisterIdea = () => {
  const [modalOpen, setModalOpen] = useState(false);

  const toggleModal = (isOpen: boolean) => {
    setModalOpen(isOpen);
  };

  return (
    <React.Fragment>
      <div className="fixed-action-btn register-idea-btn">
        <a
          className="btn-floating btn-large"
          onClick={() => toggleModal(true)}
        >
          <i className="large material-icons">add</i>
        </a>
      </div>
      {modalOpen && (
        <RegisterDialog isOpen={modalOpen} onCancel={toggleModal} />
      )}
    </React.Fragment>
  );
};

export default RegisterIdea;
