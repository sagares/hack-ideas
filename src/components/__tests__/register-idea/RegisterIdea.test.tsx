/**
 * @jest-environment jsdom
 */
import React from "react";
import { act, fireEvent, render } from "@testing-library/react";
import RegisterIdea from "../../register-idea";

describe("Register Idea Component", () => {
  let element, component;

  beforeEach(() => {
    element = document.createElement("div");

    component = render(<RegisterIdea />, {
      container: document.body.appendChild(element),
    });
  });

  afterEach(() => {
    document.body.removeChild(element);
  });

  test("renders", () => {
    expect(component.container).toMatchSnapshot();
  });

  test("opens register dialog on click of floating button", () => {
    const registerButton = element.querySelector(
      ".register-idea-btn a.btn-floating"
    );
    expect(element.querySelector(".register-idea--modal.open")).toBe(null);

    act(() => {
      fireEvent.click(registerButton);
    });

    expect(element.querySelector(".register-idea--modal.open")).toBeDefined();
  });
});
