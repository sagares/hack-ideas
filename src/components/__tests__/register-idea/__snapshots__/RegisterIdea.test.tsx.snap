// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Idea Component renders 1`] = `
<div>
  <div
    class="fixed-action-btn register-idea-btn"
  >
    <a
      class="btn-floating btn-large"
    >
      <i
        class="large material-icons"
      >
        add
      </i>
    </a>
  </div>
</div>
`;
